src/ui/mainwindow.hh

```diff
@@ -85,6 +86,8 @@ private:
 
   Ui::MainWindow ui;
 
+  StateManager stateManager;
```

Comment:
Nitpicking: Can we name it uiStateManager 😅?

Reply:
stateManager is shorter

Reply:
The next reader will be confused because, only until he opens up the class, he will know this is a UI state manager rather than a generic states manager. 

The history/favorites suffer the same issue that abnormal exit won't save them. What if we refactor the code in the same fashion, and we need a new stateMangaer 😅

Reply:
the files are placed under  ui/state_manager.hh